src/sim_engine/blockdev.rs

```diff
@@ -53,6 +60,10 @@ impl Dev for SimDev {
             None => return String::from("Conversion Failure"),
         }
     }
+
+    fn eq(&self, other: &Path) -> bool {
```

Comment:
As above, this is not any definition of equality. It includes an ```unwrap()``` which is bad, and conflicts with Rust's own definition of equality. I think it might be possible to have SimDev implement PartialEq and Eq correctly and go from there. Dev cannot implement PartialEq, because it cannot be Sized.

I already tried to have